snykTask/src/task-args.ts

```diff
@@ -150,6 +150,19 @@ class TaskArgs {
       const errorMsg = `If set, codeSeverityThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
       throw new Error(errorMsg);
     }
+
+    // code output json dependent on tested severity threshold but does not describe issues by its severity.
+    if (
+      taskTestType === TestType.CODE &&
+      ((this.codeSeverityThreshold &&
+        this.failOnThreshold &&
+        this.codeSeverityThreshold.toLowerCase() !==
+          this.failOnThreshold.toLowerCase()) ||
+        (!this.codeSeverityThreshold && this.failOnThreshold !== Severity.LOW))
+    ) {
+      const errorMsg = `If set, failOnThreshold must be matching its codeSeverityThreshold.`;
```

Comment:
question: does it need to match the value exactly (`failOnThreshold === codeSeverityThreshold`)? Or does `failOnThreshold >= codeSeverityThreshold`?

Reply:
For Snyk code, it needs to match exactly. Because there is no way to determine `>=` i.e. a more severe issue once at the `snyk code` cli output json file (unlike `snyk test` json with a severity property for each issue)

Reply:
I was referring to the general `failOnThreshold` behaviour rather than the specific validation behaviour, but we've addressed this in the other comments below.